src/parse.rs

```diff
@@ -71,6 +71,31 @@ where
     }
 }
 
+/// Parse an expected single `T` Response with `F` and store it in `into`.
+/// Responses other than `T` go into the `unsolicited` channel.
+///
+/// If zero or more than one `T` is found then into will be `None`
+pub(crate) fn parse_until_done<'input, T, F>(
+    input: &'input [u8],
+    into: &mut Option<T>,
+    unsolicited: &mut mpsc::Sender<UnsolicitedResponse>,
+    map: F,
+) -> Result<()>
+where
+    F: FnMut(Response<'input>) -> Result<MapOrNot<'input, T>>,
+{
+    let mut temp_output = Vec::<T>::new();
+
+    parse_many_into(input, &mut temp_output, unsolicited, map)?;
+
+    *into = match temp_output.len() {
+        1 => Some(temp_output.remove(0)),
+        _ => None,
```

Comment:
Any particular reason not to turn this into an error right here?

Reply:
So it leaves the decision up to the caller of whether it is an error (e.g. for cases where 0 or 1 response are acceptable).

Reply:
But they can't tell from this whether it's 0 or >1. I think it's okay to have it return an error on != 1, and then the caller can always recover the `Vec` from the error if they really want to.